tiledb/sm/subarray/subarray.cc

```diff
@@ -420,6 +537,76 @@ Status Subarray::add_range_var_by_name(
   return add_range_var(dim_idx, start, start_size, end, end_size);
 }
 
+void Subarray::get_label_range(
+    const std::string& label_name,
+    uint64_t range_idx,
+    const void** start,
+    const void** end,
+    const void** stride) const {
+  auto dim_idx = array_->array_schema_latest()
+                     .dimension_label_reference(label_name)
+                     .dimension_id();
+  if (!label_range_subset_[dim_idx].has_value() ||
+      label_range_subset_[dim_idx].name() != label_name) {
```

Comment:
Again, maybe I'm missing something w/ this abstraction, but why do we need to store and check the name here? `label_range_subset_` only has one slot per dimension, so shouldn't `optional<RangeSetAndSuperset>` plus `has_value()` here for a given dimension idx be sufficient here? (with some validation possibly done on insertion)

Reply:
This will trigger if you have two dimension labels, say 'x' and 'y', defined on the same dimension in an array, and you set ranges with dimension 'x' then try to get them back for dimension 'y'.